### PR TITLE
Release/3.12.6

### PR DESCRIPTION
**Pull request by aboedo**
Objective-C port of https://github.com/RevenueCat/purchases-ios/pull/879

### 3.12.6
- Updates `offeringsWithCompletionBlock:` to fix a case where if the backend response was erroneous, the completion block would not be called. 
- Also updates `offeringsWithCompletionBlock:` so that if there are no offerings in the RevenueCat dashboard, or no `SKProduct`s could be fetched with the product identifiers registered in the RevenueCat dashboard, the method returns an error with instructions on how to fix the issues.
    https://github.com/revenuecat/purchases-ios/pulls/885

### CODE REVIEW

**Comment by beylmk** [comment]
maybe this doesn't matter, but i ended up merging into/releasing off of the RCv4_hotfix branch

File: PurchasesTests/Mocks/MockBackend.swift [hunk 1]
```diff
@@ -64,7 +64,6 @@ class MockBackend: RCBackend {
     var stubbedGetSubscriberDataPurchaserInfo: Purchases.PurchaserInfo? = nil
     var stubbedGetSubscriberDataError: Error? = nil
 
-
```
Comment (taquitos) [inline PurchasesTests/Mocks/MockBackend.swift:67]: Little clean up, ha

**Review by taquitos — APPROVED** [review]
❌🐐🐐❌

**Review by vegaro — APPROVED** [review]

**Comment by aboedo** [comment]
update here: 
purchases-ios 3.12.6 has been released, closing this PR now since it's a hotfix.

**Pull request closed without merging**